Kickstarter-iOS/Views/Controllers/PaymentMethodsViewController.swift

```diff
@@ -182,8 +182,17 @@ extension PaymentMethodsViewController: PaymentMethodsFooterViewDelegate {
 }
 
 extension PaymentMethodsViewController: AddNewCardViewControllerDelegate {
-  internal func presentAddCardSuccessfulBanner(_ message: String) {
-    self.viewModel.inputs.cardAddedSuccessfully(message)
+  func addNewCardViewController(_ viewController: AddNewCardViewController,
+                                didSucceedWithMessage message: String) {
+    self.dismiss(animated: true) {
+      self.viewModel.inputs.cardAddedSuccessfully(message)
+    }
+  }
+
+  func addNewCardViewControllerDismissed(_ viewController: AddNewCardViewController) {
```

Comment:
Just wondering, why do we need to refresh the table view when the "Add Card" screen is dismissed but no new card has been added? Would it not be better to only refresh the table view on on 1) `viewDidLoad` 2) when a new card has been added (via the `cardAddedSuccessfully` input).

Reply:
My reasoning was that since we don't have pull to refresh or any other technique to refresh to payment methods screen (other than presenting it)...it's more useful for now to refresh both when dismissing and submitting the modal. This will for now help in situations where you'd modified your cards (on the web / other mobile client) while the modal was on and then dismissed the screen.